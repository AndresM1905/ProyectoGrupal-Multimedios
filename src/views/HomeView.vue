<script setup>
import Carousel from '../components/Carousel.vue'
import { useListsStore } from '../stores/lists'
import { computed } from 'vue'

const lists = useListsStore()

// Mostrar último añadido primero
const watchlistItems = computed(() => [...lists.watchlist].slice().reverse())
const watchedItems = computed(() => [...lists.watched].slice().reverse())
const favoritesItems = computed(() => [...lists.favorites].slice().reverse())
</script>

<template>
  <section class="section">
    <h3 class="section-title"><i class="fa-solid fa-bookmark"></i> Mi Watchlist</h3>
    <Carousel v-if="watchlistItems.length" :shows="watchlistItems" />
    <p v-else>No tienes contenido en tu Watchlist.</p>
  </section>

  <section class="section">
    <h3 class="section-title"><i class="fa-solid fa-eye"></i> Vistos</h3>
    <Carousel v-if="watchedItems.length" :shows="watchedItems" />
    <p v-else>Aún no marcaste contenido como visto.</p>
  </section>

  <section class="section">
    <h3 class="section-title"><i class="fa-solid fa-heart"></i> Favoritos</h3>
    <Carousel v-if="favoritesItems.length" :shows="favoritesItems" />
    <p v-else>No tienes contenido favorito aún.</p>
  </section>
</template>
