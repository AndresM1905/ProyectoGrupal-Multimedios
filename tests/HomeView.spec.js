import { describe, it, expect } from 'vitest'
import { mount, shallowMount } from '@vue/test-utils'
import { createTestingPinia } from '@pinia/testing'
import { setActivePinia } from 'pinia'
import HomeView from '../src/views/HomeView.vue'

function factory (listsState) {
  const pinia = createTestingPinia({
    initialState: {
      lists: listsState
    },
    stubActions: true
  })
  setActivePinia(pinia)
  return shallowMount(HomeView, {
    global: {
      plugins: [pinia],
      stubs: {
        Carousel: true // evita renderizar carrusel completo
      }
    }
  })
}

describe('HomeView snapshot', () => {
  it('renderiza secciones vacías como esperado', () => {
    const wrapper = factory({ watchlist: [], watched: [], favorites: [] })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renderiza con ítems en las listas', () => {
    const dummy = [{ id: 1, title: 'Fake', img: 'a.jpg', type: 'serie' }]
    const wrapper = factory({ watchlist: dummy, watched: [], favorites: dummy })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
