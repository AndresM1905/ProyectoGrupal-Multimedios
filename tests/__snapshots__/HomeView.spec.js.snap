// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomeView snapshot > renderiza con ítems en las listas 1`] = `
"<section class="section">
  <h3 class="section-title"><i class="fa-solid fa-bookmark"></i> Mi Watchlist</h3>
  <carousel-stub shows="[object Object]"></carousel-stub>
</section>
<section class="section">
  <h3 class="section-title"><i class="fa-solid fa-eye"></i> Vistos</h3>
  <p>Aún no marcaste contenido como visto.</p>
</section>
<section class="section">
  <h3 class="section-title"><i class="fa-solid fa-heart"></i> Favoritos</h3>
  <carousel-stub shows="[object Object]"></carousel-stub>
</section>"
`;

exports[`HomeView snapshot > renderiza secciones vacías como esperado 1`] = `
"<section class="section">
  <h3 class="section-title"><i class="fa-solid fa-bookmark"></i> Mi Watchlist</h3>
  <p>No tienes contenido en tu Watchlist.</p>
</section>
<section class="section">
  <h3 class="section-title"><i class="fa-solid fa-eye"></i> Vistos</h3>
  <p>Aún no marcaste contenido como visto.</p>
</section>
<section class="section">
  <h3 class="section-title"><i class="fa-solid fa-heart"></i> Favoritos</h3>
  <p>No tienes contenido favorito aún.</p>
</section>"
`;
